src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,12 +4465,199 @@ subroutine cat_enkf_increments(                               &
+                ! 3. Derive SWE, snow heat content, and snow depth increments for each layer from total SWE increment 
+                
+                swe_ana   = max(swe_fcst + swe_incr(kk,n_e), 0.0)    ! total SWE after analysis
+                 
+                swe_ana_array(1,1) = swe_ana   
+                
+                call StieglitzSnow_calc_asnow( 1, 1, swe_ana_array, asnow_ana_array )
+                
+                asnow_ana = asnow_ana_array(1)                       ! asnow after analysis
+
+                if (swe_fcst>=SCF_ANA_MINFCSTSWE) then
+                   swe_ratio = swe_ana / swe_fcst 
+                else
+                   swe_ratio = 1.  ! not used when swe_fcst<SCF_ANA_MINFCSTSWE but set to neutral just in case
+                end if 
+                
+                ! loop through snow layers and compute SWE, snow heat content, and snow depth analysis for each layer
+                
+                do isnow=1,N_snow
+                   
+                   if     (asnow_ana == 0.0) then  
+                      
+                      ! no snow in analysis, remove all snow
+                      
+                      tmp_wesn(kk,n_e,isnow) = 0.0
+                      tmp_htsn(kk,n_e,isnow) = 0.0
+                      tmp_sndz(kk,n_e,isnow) = 0.0 
+                      
+                   elseif (swe_fcst < SCF_ANA_MINFCSTSWE) then
+                      
+                      ! too little snow in forecast, use generic properties for added snow
+                      
+                      tmp_wesn(kk,n_e,isnow) = swe_ana / N_snow                           ! distribute SWE evenly across layers
+                      tmp_htsn(kk,n_e,isnow) = (0.0 - MAPL_ALHF)*tmp_wesn(kk,n_e,isnow)   ! assign heat content for snow at 0 deg C and without liquid water content (100% frozen) based on StieglitzSnow: htsn = (cpw*tsnow - fice*MAPL_ALHF)*swe
+                      tmp_sndz(kk,n_e,isnow) = (WEMIN / RHOFS) / N_snow                   ! assign snow depth consistent with density of freshly fallen snow (must have SCF_ANA_MAXINCRSWE<=WEMIN)
+                      
+                   else
+                      
+                      ! snow in forecast and analysis, derive properties of analysis snow from properties of forecast snow
+                      
+                      tmp_wesn(kk,n_e,isnow) = cat_progn(kk,n_e)%wesn(isnow) * swe_ratio
+                      tmp_htsn(kk,n_e,isnow) = cat_progn(kk,n_e)%htsn(isnow) * swe_ratio  ! What if fcst snow is at 0 deg C and contains liquid water?  TBD reichle 20221007, could update as:
+                      !tmp_htsn(kk, n_e, isnow)   = (cpw * tpsn(kkm n_e, isnow)  - fices(kk, n_e, isnow) * MAPL_ALHF) * tmp_wesn(kk, n_e, isnow)
+                      
+                      if (asnow_ana < 1.) then
+                         
+                         tmp_sndz(kk,n_e,isnow) = cat_progn(kk,n_e)%sndz(isnow)          ! snow depth remains constant b/c swe_ana<1.
```

Comment:
@lcandre2, @amfox37, @gmao-qliu: I'm wondering if we need another if condition here.  If asnow_ana<1 **and** asnow_fcst<1, then keeping the snow depth constant as coded makes sense.  I'm not sure what we would want to do for asnow_ana<1 **and** asnow_fcst==1.  In this case, we had full snow coverage before the analysis and SWE potentially well above 13 kg/m2, with a correspondingly deep snow pack.  In this case, removing SWE and simultaneously keeping the depth constant would presumably reduce the snow density (potentially even below the min value), but we want to keep the density constant.  I'm struggling to figure out what we would want to do instead. 
 Here's my best guess late on a Fri afternoon:
```
if (asnow_ana < 1.) then
  if (asnow_fcst < 1.) then
    tmp_sndz(kk,n_e,isnow) = cat_progn(kk,n_e)%sndz(isnow)          ! snow depth remains constant b/c swe_ana<1.
  else
    tmp_sndz(kk,n_e,isnow) = tmp_wesn(kk,n_e,isnow) / ( cat_progn(kk,n_e)%wesn(isnow) / cat_progn(kk,n_e)%sndz(isnow) ) / asnow_ana
  end if
else
  ! compute analysis snow depth from (layer-specific) forecast snow density
  tmp_sndz(kk,n_e,isnow) = tmp_wesn(kk,n_e,isnow) / ( cat_progn(kk,n_e)%wesn(isnow) / cat_progn(kk,n_e)%sndz(isnow) )
end if
```
**If** this makes any sense at all, then __maybe__ we could merge the `else` statements as follow (not 100% sure of the logic...):

```
if ( (asnow_ana < 1.) .and. (asnow_fcst < 1.) ) then
  tmp_sndz(kk,n_e,isnow) = cat_progn(kk,n_e)%sndz(isnow)          ! snow depth remains constant b/c swe_ana<1.
else
  ! compute analysis snow depth from (layer-specific) forecast snow density
  ! (need to divide by asnow_ana to get sndz within the snow-covered portion in case asnow_ana<1)  
  tmp_sndz(kk,n_e,isnow) = tmp_wesn(kk,n_e,isnow) / ( cat_progn(kk,n_e)%wesn(isnow) / cat_progn(kk,n_e)%sndz(isnow) ) / asnow_ana

end if
```
In any case, this probably requires another round of thinking.

Reply:
@gmao-rreichle, I agree that maintaining the snow depth in the situation where fcst=1 and ana<1 could lead to physically unrealistic (low) densities and the fix should be similar to the situation where fcst<1 and ana=1. Your proposed solution makes sense to me -- I would have forgotten the inclusion of asnow_ana.  The simplification is logical to me, in the fcst<1 & ana=1, the last term in depth calculation would be 1, making it identical to fcst=1 & ana<1. No point in having two else statements when 1 works.

Reply:
I am making and testing this modification now.